### DISCUSSION TITLE
SEE DELUGE AS EXTERNAL DRIVE OVER USB

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#536] [Ideas]
If we could plug the deluge directly into the computer for file transfers/edits it would save a lot of wear and tear on the SD drive.

**Comment by sapphire-arches** [comment]
Unfortunately this is more difficult than it sounds because the Deluge
itself can access the card at any time. Therefore we can not just present
the SD card as a normal external drive, as that risks corruption of the
drive contents (if we expose it directly) or requires excessive memory to
emulate a file system. The instead, we want to expose the file system
either over MTP (like Android phones) or a MIDI sysex interface.

**Reply by worker-gray** [reply]
would it expose the disk to any more corruption than inserting the disk directly into a computer?

**Reply by sapphire-arches** [reply]
Exposing it as a regular disk drive would yes. The computer and the Deluge could both end up writing to the same part of the drive at the same time and both parts would get corrupt.

**Reply by sichtbeton** [reply]
could this be solved with a "disk mode", as separate mode that the deluge can be booted to (by holding some button and power maybe, like FW update)? the op-z and op-1 do this.

**Reply by Radius9999** [reply]
Yes, I second the "disk mode".  The SD card doesn't have to be live in the Deluge.  It just has to be accessible via usb.  The Deluge could be put into a dormant stage, and the SD card mapped as a drive in windows.  The Akai Force does it this way.

**Reply by gramster** [reply]
The Force in particular has a good implementation of this because you can enter/exit it at will without a reboot in the middle of doing other things, unlike devices that require powering up in normal vs disk mode.

**Comment by jamiefaye** [comment]
The Force has a "Computer Mode" you can enter, which makes the SSD drive (if installed) available. Unfortunately both Windows and MacOS don't always write changes made via this interface out to the external drive right-away, preferring to keep a write cache in memory. You have to do a "safe removal" to shut down the USB drive. This means that if anything goes wrong anywhere, you can have a corrupted drive. This is in contrast to how the Deluge handles its SD-Card, where they make sure they "flush" any file system changes to the card immediately. This is what makes Downrush work relatively well.

You would think that MS and Apple would get-with-it and flush the data to USB on a regular basis, but they don't. A major annoyance and a cause of drive corruption.
